pkg/osvscanner/osvscanner.go

```diff
@@ -238,11 +255,118 @@ func DoScan(actions ScannerActions, r reporter.Reporter) (models.VulnerabilityRe
+		},
+	}
+
+	// --- Setup Accessors/Clients ---
+	accessors, err := initializeExternalAccessors(r, actions)
+	if err != nil {
+		return models.VulnerabilityResults{}, fmt.Errorf("failed to initialize accessors: %w", err)
+	}
+
+	// --- Initialize Image To Scan ---
+	var img *image.Image
+	if actions.ScanOCIImage != "" {
+		img, err = image.FromTarball(actions.ScanOCIImage, image.DefaultConfig())
+		r.Infof("Scanning image %q\n", actions.ScanOCIImage)
+	} else if actions.Image != "" {
+		path, exportErr := imagehelpers.ExportDockerImage(r, actions.Image)
+		if exportErr != nil {
+			return models.VulnerabilityResults{}, exportErr
+		}
+
+		// If Image is a local tar file, then path == Image, and we shouldn't remove it
+		if path != actions.Image {
+			defer os.Remove(path)
+		}
+		img, err = image.FromTarball(path, image.DefaultConfig())
+		r.Infof("Scanning image %q\n", actions.Image)
+	}
+	if err != nil {
+		return models.VulnerabilityResults{}, err
+	}
+	defer func() {
+		err := img.CleanUp()
+		if err != nil {
+			r.Errorf("Failed to clean up image: %s\n", err)
+		}
+	}()
+
+	// --- Do Scalibr Scan ---
+	scanner := scalibr.New()
+	scalibrSR, err := scanner.ScanContainer(context.Background(), img, &scalibr.ScanConfig{
+		FilesystemExtractors: scanners.BuildArtifactExtractors(),
+	})
+	if err != nil {
+		return models.VulnerabilityResults{}, fmt.Errorf("failed to scan container image: %w", err)
+	}
+
+	if len(scalibrSR.Inventories) == 0 {
+		return models.VulnerabilityResults{}, ErrNoPackagesFound
+	}
 
-		// TODO(v2): in the next breaking release of osv-scanner, consider
-		// returning a ScanError instead of an error.
+	// --- Save Scalibr Scan Results ---
+	scanResult.PackageScanResults = make([]imodels.PackageScanResult, len(scalibrSR.Inventories))
+	for i, inv := range scalibrSR.Inventories {
+		scanResult.PackageScanResults[i].PackageInfo = imodels.FromInventory(inv)
+		scanResult.PackageScanResults[i].LayerDetails = inv.LayerDetails
+	}
+
+	// --- Fill Image Metadata ---
+	scanResult.ImageMetadata, err = imagehelpers.BuildImageMetadata(img, accessors.BaseImageMatcher)
+	if err != nil { // Not getting image metadata is not fatal
+		r.Errorf("Failed to fully get image metadata: %v", err)
+	}
+
+	// ----- Filtering -----
+	filterUnscannablePackages(r, &scanResult)
+
+	// --- Make Vulnerability Requests ---
+	if accessors.VulnMatcher != nil {
+		err = makeVulnRequestWithMatcher(r, scanResult.PackageScanResults, accessors.VulnMatcher)
+		if err != nil {
+			return models.VulnerabilityResults{}, err
+		}
+	}
+
+	// --- Make License Requests ---
+	if accessors.LicenseMatcher != nil {
+		err = accessors.LicenseMatcher.MatchLicenses(context.Background(), scanResult.PackageScanResults)
+		if err != nil {
+			return models.VulnerabilityResults{}, err
+		}
+	}
+
+	results := buildVulnerabilityResults(r, actions, &scanResult)
+
+	filtered := filterResults(r, &results, &scanResult.ConfigManager, actions.ShowAllPackages)
```

Comment:
Do we want to filter out vulnerabilities from `usr/` directories here?

Reply:
Added annotations to the under `usr/` directory.